mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputFormat.java

```diff
@@ -19,21 +19,445 @@
+          throw new UnsupportedOperationException("Cannot write format: " + fileFormat);
+      }
+      LOG.info("IcebergRecordWriter is created in {} with {}", location, fileFormat);
+    }
+
+    @Override
+    public void write(Writable row) {
+      Preconditions.checkArgument(row instanceof IcebergWritable);
+
+      // TODO partition handling
+      appender.add(((IcebergWritable) row).record());
+    }
+
+    @Override
+    public void write(NullWritable key, IcebergWritable value) {
+      write(value);
+    }
+
+    @Override
+    public void close(boolean abort) throws IOException {
+      appender.close();
+      if (!abort) {
+        fileData.put(taskAttemptId, new ClosedFileData(location, fileFormat, appender.length(), appender.metrics()));
+      }
+    }
+
+    @Override
+    public void close(org.apache.hadoop.mapreduce.TaskAttemptContext context) throws IOException {
+      close(false);
+    }
+
+    @Override
+    public void close(Reporter reporter) throws IOException {
+      close(false);
+    }
+  }
+
+  /**
+   * A dummy committer - not related to the Hive transactions.
+   */
+  public static final class IcebergOutputCommitter extends OutputCommitter {
+
+    @Override
+    public void setupJob(JobContext jobContext) {
+      // do nothing.
+    }
+
+    @Override
+    public void setupTask(TaskAttemptContext taskAttemptContext) {
+      // do nothing.
+    }
+
+    @Override
+    public boolean needsTaskCommit(TaskAttemptContext taskAttemptContext) {
+      return true;
+    }
+
+    @Override
+    public void commitTask(TaskAttemptContext context) throws IOException {
+      ClosedFileData closedFileData = fileData.remove(context.getTaskAttemptID());
+      String commitFileLocation = generateCommitFileLocation(context.getJobConf(), context.getTaskAttemptID());
+
+      // If we created a new data file, then create the committed file for this
+      if (closedFileData != null) {
+        createCommittedFileFor(new HadoopFileIO(context.getJobConf()), closedFileData, commitFileLocation);
+      } else {
+        TaskType taskType = context.getTaskAttemptID().getTaskID().getTaskType();
+        boolean isWrite = context.getJobConf().getBoolean(HiveIcebergStorageHandler.WRITE_KEY, false);
+        boolean mapOnly = context.getJobConf().getNumReduceTasks() == 0;
+
+        // If we writing and the task is either reducer or a map in a map-only job then write an empty commit file
+        if (isWrite && (TaskType.REDUCE.equals(taskType) || (TaskType.MAP.equals(taskType) && mapOnly))) {
+          createCommittedFileFor(new HadoopFileIO(context.getJobConf()), new ClosedFileData(), commitFileLocation);
+        }
+      }
+    }
+
+    @Override
+    public void abortTask(TaskAttemptContext context) {
+      FileIO io = new HadoopFileIO(context.getJobConf());
+
+      // Clean up local cache for metadata
+      fileData.remove(context.getTaskAttemptID());
+
+      // Remove the result file for the failed task
```

Comment:
Shouldn't this try to read the result file and delete all of the data files if it is present?

Reply:
OutputCommitter.abortTask() is only for cleaning up the files generated by the task which is failed.
The job might continue with a 2nd attempt on the task, so we do not want to clean up the other files which should be done in the OutputCommitter.abortJob().

I was concentrating on the happy path here, but I was so enthusiastic when I was playing with the Tasks construct that I accidentally implemented abortTask() too 😄 

Do you think the other parts of the patch are solid enough now that we should move forward to implement the error handling path?

Reply:
Added the abortTask handling